profile/profile.go

```diff
@@ -16,9 +17,22 @@ type payloadIdentifier struct {
 }
 
 func (mc *Mobileconfig) GetPayloadIdentifier() (string, error) {
+	mcBytes := *mc
+	if len(mcBytes) > 5 && string(mcBytes[0:5]) != "<?xml" {
+		p7, err := pkcs7.Parse(mcBytes)
+		if err != nil {
+			return "", errors.Wrapf(err, "Mobileconfig is not XML nor PKCS7 parseable")
+		}
+		err = p7.Verify()
```

Comment:
This is similar in the OpenSSL world, too — verifying the signature cryptographically verifies against the data (but not the included certificates, or where they're from, or anything like that).